numba/parfors/parfor.py

```diff
@@ -1836,7 +1838,9 @@ def _numpy_to_parfor(self, equiv_set, lhs, expr):
         if call_name in ['zeros', 'ones'] or mod_name == 'numpy.random':
             return self._numpy_map_to_parfor(equiv_set, call_name, lhs, args, kws, expr)
         # return error if we couldn't handle it (avoid rewrite infinite loop)
-        raise NotImplementedError("parfor translation failed for ", expr)
+        raise errors.RewriteUnsupportedError(
+            f"parfor translation failed for {expr}", loc=exprloc,
```

Comment:
Should that be `expr.loc`? If so, no build failures suggests this isn't tested?

Reply:
good catch. some of the exception raise in Parfor are not covered by the testsuite

Reply:
not sure how to trigger this one.